routes/api/contacts.js

```diff
@@ -1,25 +1,84 @@
-const express = require('express')
+const {
+  listContacts,
+  getContactById,
+  addContact,
+  removeContact,
+  updateContact,
+} = require("../../models/contacts");
 
-const router = express.Router()
+const express = require("express");
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+const Joi = require("joi");
+const schema = Joi.object({
+  name: Joi.string().alphanum().min(2).max(30),
+  email: Joi.string().email({ minDomainSegments: 2, multiple: true }),
+  phone: Joi.string().alphanum().min(9).max(15),
+});
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+const router = express.Router();
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
+  try {
+    const contacts = await listContacts();
+    res.json(contacts);
+  } catch (error) {
+    console.log(error);
+  }
+});
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:contactId", async (req, res, next) => {
+  try {
+    const id = req.params.contactId;
+    const searchedContact = await getContactById(id);
+    searchedContact
+      ? res.status(200).json(searchedContact)
+      : res.status(404).json({ message: "Not found" });
+  } catch (error) {
+    console.log(error);
+  }
+});
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post("/", async (req, res, next) => {
+  try {
+    const validateResult = schema.validate(req.body);
+    if (validateResult.error || !req.body)
+      return res.status(400).json({ message: "missing required name - field" });
 
-module.exports = router
+    const newContact = await addContact(req.body);
+    res.status(201).json({ newContact });
+  } catch (error) {
+    console.log(error);
+  }
+});
+
+router.delete("/:contactId", async (req, res, next) => {
+  try {
+    const id = req.params.contactId;
+    const result = await removeContact(id);
+
+    if (!result) res.status(404).json({ message: "Not found" });
+    if (result) res.status(200).json({ message: result });
+  } catch (error) {
+    console.log(error);
+  }
+});
+
+router.put("/:contactId", async (req, res, next) => {
+  try {
+    const id = req.params.contactId;
+    const body = req.body;
+
+    const validateResult = schema.validate(body);
+    if (validateResult.error || !body)
+      return res.status(400).json({ message: "missing fields" });
+
+    const result = await updateContact(id, body);
+
+    if (!result) res.status(404).json({ message: "Not found" });
+    if (result) res.status(200).json({ message: result });
+  } catch (error) {
+    console.log(error);
```

Comment:
```suggestion
next(error)
```

Reply:
Albo możesz przekazać jakąś własną wiadomość, co byłoby lepsze. Natomiast na końcu bloku try catch w endpoincie to chcemy przekazać użytkownikowic co jest nie tak. console.log jest dla Cie bie do testowania.

Reply:
Next przekaże kontrolę do pliku app.js tam gdzie jest uzyty middleware z pierwszym elementem err, tam zostanie błąd przechwycony. Więc tak nie musisz go używać ale trzeba coś zwrócić użytkownikowi więc res.status(błędu).send(wiadomość)